pkg/cloudinit/controlplane_init_test.go

```diff
@@ -40,23 +41,127 @@ func TestNewInitControlPlane(t *testing.T) {
 				Owner:       "root:root",
 			}},
 			ConfigFileContents:  "### config file ###",
-			MicroclusterAddress: ":2380",
+			MicroclusterAddress: "10.0.0.10",
 		},
+		Token:              "test-token",
+		K8sdProxyDaemonSet: "test-daemonset",
 	})
 
-	// TODO: add tests for expected files and commands
-	g.Expect(err).To(BeNil())
+	g.Expect(err).ToNot(HaveOccurred())
+
+	// Verify the boot commands.
+	g.Expect(config.BootCommands).To(Equal([]string{"bootcmd"}))
+
+	// Verify the run commands.
 	g.Expect(config.RunCommands).To(Equal([]string{
 		"set -x",
 		"prerun1",
 		"prerun2",
 		"/capi/scripts/install.sh",
 		"/capi/scripts/bootstrap.sh",
+		"/capi/scripts/load-images.sh",
 		"/capi/scripts/wait-apiserver-ready.sh",
 		"/capi/scripts/deploy-manifests.sh",
 		"/capi/scripts/configure-token.sh",
 		"/capi/scripts/create-sentinel-bootstrap.sh",
 		"postrun1",
 		"postrun2",
 	}))
+
+	// Define the expected files to write with their content, path, permissions, and owner.
+	expectedWriteFiles := []cloudinit.File{
+		{
+			Path:        "/tmp/file",
+			Content:     "test file",
+			Permissions: "0400",
+			Owner:       "root:root",
+		},
+		{
+			Path:        "/capi/etc/config.yaml",
+			Content:     "### config file ###",
+			Permissions: "0400",
+			Owner:       "root:root",
+		},
+		{
+			Path:        "/capi/etc/microcluster-address",
+			Content:     "10.0.0.10:2380",
+			Permissions: "0400",
+			Owner:       "root:root",
+		},
+		{
+			Path:        "/capi/etc/snap-track",
+			Content:     "1.30-classic/stable",
+			Permissions: "0400",
+			Owner:       "root:root",
+		},
+		{
+			Path:        "/capi/etc/token",
+			Content:     "test-token",
+			Permissions: "0400",
+			Owner:       "root:root",
+		},
+		{
+			Path:        "/capi/manifests/00-k8sd-proxy.yaml",
+			Content:     "test-daemonset",
+			Permissions: "0400",
+			Owner:       "root:root",
+		},
+	}
+
+	scriptFiles := map[string]string{
+		"./scripts/install.sh":                   "/capi/scripts/install.sh",
+		"./scripts/bootstrap.sh":                 "/capi/scripts/bootstrap.sh",
+		"./scripts/load-images.sh":               "/capi/scripts/load-images.sh",
+		"./scripts/join-cluster.sh":              "/capi/scripts/join-cluster.sh",
+		"./scripts/wait-apiserver-ready.sh":      "/capi/scripts/wait-apiserver-ready.sh",
+		"./scripts/deploy-manifests.sh":          "/capi/scripts/deploy-manifests.sh",
+		"./scripts/configure-token.sh":           "/capi/scripts/configure-token.sh",
+		"./scripts/create-sentinel-bootstrap.sh": "/capi/scripts/create-sentinel-bootstrap.sh",
+	}
```

Comment:
this test does not protect us from any regressions. as such, it's best to write it differently (e.g. that there is a cloudinit.File with Path /capi/scripts/install.sh, bootstrap.sh, etc.

As written, this is just something that we will be updating along with the implementation, so offers very little value

this is okay but we are not really testing anything with it. it is enough to check that the files are in the writeFiles.


this is because, if we ever add any new files, this test breaks by definition. it does not prevent any regressions